1. Basic Web Development/script.ts

```diff
@@ -0,0 +1,405 @@
+            submitEdit(form);
+        }
+    });
+}
+
+function submitPermissions(form: HTMLElement) {
+    let messageData: { id: string, status: boolean }[] = [];
+
+    this.PERMISSIONS.forEach((element: Permission) => {
+        let checkedStatus = (document.getElementById(element.id) as HTMLInputElement).checked;
+
+        let newData = { "id": element.id, "status": checkedStatus };
+
+        messageData.push(newData);
+    });
+
+    const messageJSON = JSON.stringify(messageData);
+
+    fetch(API_URL, { method: 'POST', headers: HEADERS, body: messageJSON })
+        .then(response => response.json())
+        .then(data => {
+            console.log(data);
+        })
+        .catch((error) => {
+            console.error('Error:', error);
+            permissionsRoutine();
+        });
+}
+
+function submitCreation(form: HTMLElement) {
+    const type = document.querySelector('input[type=radio][name=presetC]:checked').id;
+    const text = (document.getElementById('permName') as HTMLInputElement).value;
+
+    if (!text) {
+        document.getElementById('text-alert').classList.remove('no-display');
+        return;
+    } else {
+        document.getElementById('text-alert').classList.add('no-display');
+    }
+
+    const newPermission = new Permission(text, type)
+
+    this.PERMISSIONS.push(newPermission);
+
+    fetch(API_URL + '/pref/create', { method: 'POST', headers: HEADERS, body: JSON.stringify(newPermission) })
+        .then(response => response.json())
+        .then(data => {
+            console.log(data);
+        })
+        .catch((error) => {
+            console.error('Error:', error);
+            permissionsRoutine();
+        });
+
+    createPermissionsForm(form);
+}
+
+function submitEdit(form: HTMLElement) {
+    const type = document.querySelector('input[type=radio][name=presetE]:checked').id;
+    const text = (document.getElementById('permNameEdit') as HTMLInputElement).value;
+
+    if (!text) {
+        document.getElementById('text-alert-edit').classList.remove('no-display');
+        return;
+    } else {
+        document.getElementById('text-alert-edit').classList.add('no-display');
+    }
+
+    const index = this.PERMISSIONS.findIndex((perm: Permission) => perm.id == this.NOWEDITING.id);
+
+    this.PERMISSIONS[index].type = type.replace('Edit', '');
+    this.PERMISSIONS[index].text = text;
+
+    document.getElementById('edit-form').classList.add('no-display');
+
+    const messageJSON = JSON.stringify({index: index, type: type.replace('Edit', ''), text: text});
+
+    fetch(API_URL + '/pref/update', { method: 'PUT', headers: HEADERS, body: messageJSON })
+        .then(response => response.json())
+        .then(data => {
+            console.log(data);
+        })
+        .catch((error) => {
+            console.error('Error:', error);
+            permissionsRoutine();
+        });
+
+    createPermissionsForm(form);
+}
+
+function handleLock(lock: HTMLInputElement, popUp: Element) {
```

Comment:
handleLock and handlePopUp are a bit confusing for someone to follow, specially since they are working on the same element. Maybe you should consider a refactor here.